# Создать  и активировать  виртуальное окружение
'''
python3 -m venv venv
source venv/bin/activate
deactivate
'''
# Добавить название вашего виртуального окружения venv в  исключения .gitignore
'''
Добавилено название .venv/ в .gitignore
'''
# Создать пакет Python добавить стороннюю библиотеку
'''
Скачал django, fastapi. 
Использовал команды 
pip list
pip install
pip unistall
pip list | grap fastapi
'''
# Загрузить пакет на Pypi
'''
Статья полезная с описанием действий.
https://towardsdatascience.com/how-to-upload-your-python-package-to-pypi-de1b363a1b3

Мой пакет залитый на Pypi
С первого раза не залился, так имя такое уже существовало, пришлось изменять имя в фале setup.py 
и после заного создавать dist.

pip install Guess-my-number-EVG==0.1

Был баг запуска отпишусь о решении: 
1. Запускать нужно всегда напрямую через код, типа python3 dz13.py;
2. При создании pip указывал папку название test, потом после ее установки через pip пришлось искать в папках и изменять на evgeniy, тогда все получилось;
3. from evgeniy import run  - так я писал для запуска моего скрипта в файле dz13.py
4. в корневой папке где находился файл dz13.py должен был находиться файл __init_ .py - создавал его;
5. Ну так как я запускал все это с виртуального окружения, нашел и поменял название папки test на evgeniy путь следующий: venv/lib/python3.9/site-packages

'''

# Почитать про PEP8 и PEP20